lib/cocoapods-downloader/git.rb

```diff
@@ -77,6 +77,10 @@ def clone(force_head = false, shallow_clone = true)
         ui_sub_action('Git download') do
           begin
             git! clone_arguments(force_head, shallow_clone)
+            if !force_head && !validate_tag
+              target_git 'checkout', '--quiet', "tags/#{options[:tag]}"
```

Comment:
This might require also fetching the tag? eg refs/tags/...

Reply:
yea definitely, need to figure out how to successfully test that